substrate/frame/collective/src/lib.rs

```diff
@@ -1083,24 +1073,37 @@ impl<T: Config<I>, I: 'static> ChangeMembers<T::AccountId> for Pallet<T, I> {
 		// remove accounts from all current voting in motions.
 		let mut outgoing = outgoing.to_vec();
 		outgoing.sort();
-		for h in Self::proposals().into_iter() {
-			<Voting<T, I>>::mutate(h, |v| {
-				if let Some(mut votes) = v.take() {
-					votes.ayes = votes
-						.ayes
-						.into_iter()
-						.filter(|i| outgoing.binary_search(i).is_err())
-						.collect();
-					votes.nays = votes
-						.nays
-						.into_iter()
-						.filter(|i| outgoing.binary_search(i).is_err())
-						.collect();
-					*v = Some(votes);
-				}
-			});
-		}
-		Members::<T, I>::put(new);
+		<Voting<T, I>>::translate_values(
+			|mut votes: Votes<
+				T::AccountId,
+				frame_system::pallet_prelude::BlockNumberFor<T>,
+				T::MaxMembers,
+			>| {
+				votes.ayes = votes
+					.ayes
+					.into_iter()
+					.filter(|i| outgoing.binary_search(i).is_err())
+					.collect::<Vec<T::AccountId>>()
```

Comment:
Ah it's impossible to collect directly back into a `BoundedVec`? That seems like an oversight if not...

Reply:
It is impossible. I will remove the expect by using a `truncate_from` instead, which I discovered later